### PR TITLE
allow usage of PRStatus.merged for Github get_pr_list(status)

### PR DESCRIPTION
**Pull request by marusinm**
Based on our discussion in PR #128 this code makes consistency in `get_pr_list(status)` between Pagure and Github. We can use `PRStatus.merged` for Github now.

Can we close #128 without merging? It was easier to create a new branch since the code is completely different.

### CODE REVIEW

File: ogr/services/github.py [hunk 1]
```diff
@@ -358,8 +358,17 @@ def add_issue_labels(self, issue_id, labels) -> None:
...
+            direction="desc",
         )
+
+        if status == PRStatus.merged:
+            prs = list(prs)  # Github PaginatedList into list()
+            for pr in prs:
+                if not pr.is_merged():  # parse just merged PRs
+                    prs.remove(pr)
         try:
             return [self._pr_from_github_object(pr) for pr in prs]
```
Comment (lachmanfrantisek) [inline ogr/services/github.py]: I think we also need to change the `_pr_from_github_object` method to translate the status correctly. Or am I wrong?

e.g.:
```python
get_pr_list(status=PRStatus.all)
```

Here we get the `closed` state for all `merged`.

Comment (marusinm) [reply ogr/services/github.py]: Yes, I was thinking about what is better, the decision is on us. I'll change it.

Comment (lachmanfrantisek) [reply ogr/services/github.py]: Your code is correct to this point, we only need to extend `_pr_from_github_object` to differentiate the closed and merged status.

File: ogr/services/github.py [hunk 2]
```diff
@@ -593,7 +593,9 @@ def _pr_from_github_object(github_pr: GithubPullRequest) -> PullRequest:
         return PullRequest(
             title=github_pr.title,
             id=github_pr.number,
-            status=PRStatus[github_pr.state],
+            status=PRStatus.merged
+            if github_pr.is_merged()
+            else PRStatus[github_pr.state],
```
Comment (lachmanfrantisek) [inline ogr/services/github.py]: :+1:

**Comment by lachmanfrantisek** [comment]
@marusinm thanks for the pull request. I'll merge that after the completion of the tests.

**Comment by rpitonak** [comment]
pre-commit-black failed unfortunately

**Comment by lachmanfrantisek** [comment]
Oh, the **pre-commit** does not like your last last commit:

```python
13:23:56  Check Yaml...............................................................Passed
13:23:56  Detect Private Key.......................................................Passed
13:23:56  Fix End of Files.........................................................Passed
13:23:56  Trim Trailing Whitespace.................................................Passed
13:23:56  Flake8...................................................................Passed
13:23:57  black....................................................................Failed
13:23:57  hookid: black
13:23:57  
13:23:57  Files were modified by this hook. Additional output:
13:23:57  
13:23:57  reformatted /root/ogr/services/github.py
13:23:57  All done! ✨ 🍰 ✨
13:23:57  1 file reformatted, 29 files left unchanged.
13:23:57  
13:24:11  mypy.....................................................................Passed
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }
[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
```

I think the last commit is not needed.

**Comment by marusinm** [comment]
@lachmanfrantisek then I maybe didn't understand your last review comment. I changed just one line which looks better. If I repair it with pre-commit I can just go one commit back. Please look at [my last change](https://github.com/packit-service/ogr/pull/130/commits/cf86339b0eaef1c6e6a5c5668a0868da076ffe04).

**Comment by rpitonak** [comment]
@marusinm yes your last commit was not needed, Frantisek's last review was just "thumbs up"

**Comment by lachmanfrantisek** [comment]
> @lachmanfrantisek then I maybe didn't understand your last review comment. I changed just one line which looks better. If I repair it with pre-commit I can just go one commit back. Please look at [my last change](https://github.com/packit-service/ogr/pull/130/commits/cf86339b0eaef1c6e6a5c5668a0868da076ffe04).

@marusinm That was just to let you know it's ok and that I will release a new version after merging this PR.

Just remove the last commit and we are ready to merge.

**Comment by marusinm** [comment]
😄Sorry, misunderstanding on my side. I deleted the last commit.

**Pull request merged**